lib/PrintAsClang/PrimitiveTypeMapping.cpp

```diff
@@ -19,29 +19,78 @@
 
 using namespace swift;
 
+/// Find the implementation of the named type in the named module if loaded.
+static TypeDecl *findTypeInModuleByName(ASTContext &ctx,
+                                        Identifier moduleName,
+                                        Identifier typeName) {
+  auto module = ctx.getLoadedModule(moduleName);
+  if (!module)
+    return nullptr;
+
+  // Find all of the declarations with this name in the Swift module.
+  SmallVector<ValueDecl *, 1> results;
+  module->lookupValue(typeName, NLKind::UnqualifiedLookup, results);
```

Comment:
In what cases do we expect to see multiple results from the lookup?

Reply:
In practice, I would expect to find at most one type definition for a given name. In theory, I think we could find two different private types with the same name in distinct files. Since we control the modules named here I think it's unlikely we'd see this, we already expect the types to be public so it should avoid the duplicates.

Reply:
It makes sense, I added the assert.